src/main/java/io/sealights/plugins/sealightsjenkins/buildsteps/cli/SealightsCLIBuildStep.java

```diff
@@ -33,6 +36,44 @@ public SealightsCLIBuildStep(boolean enabled, boolean failBuildIfStepFail,
         this.cliRunner = cliRunner;
     }
 
+    /* * The goal of this method is to support migration of data between versions
+    * of this plugin.
+    */
+    private Object readResolve() {
+        if (cliRunner != null) {
+            return resolveFromCLiRunner();
+        }
+        return this;
+    }
+
+    private Object resolveFromCLiRunner() {
+        StringBuilder  additionalArgs = new StringBuilder();
+        if (this.commandMode instanceof CommandMode.ConfigView) {
+            if (!StringUtils.isNullOrEmpty(cliRunner.getAppName()))
+                ((CommandMode.ConfigView) commandMode).setAppName(cliRunner.getAppName());
+            if (!StringUtils.isNullOrEmpty(cliRunner.getBranchName()))
+                ((CommandMode.ConfigView) commandMode).setBranchName(cliRunner.getBranchName());
+            if (cliRunner.getBuildName() != null)
+                ((CommandMode.ConfigView) commandMode).setBuildName(cliRunner.getBuildName());
```

Comment:
The code is a bit hard to read.
Maybe you can cast the object once and use it in all places?.